pkg/mutation/path/tester/tester.go

```diff
@@ -71,16 +77,35 @@ func ValidatePathTests(location *parser.Path, pathTests []Test) error {
 
 // New creates a new Tester object
 func New(tests []Test) (*Tester, error) {
+	paths := make(map[int]*parser.Path)
 	idx := &Tester{
 		tests: make(map[int]Condition),
 	}
+	// Read in all tests before checking for conflicts.
+	for _, test := range tests {
+		i := len(test.SubPath.Nodes) - 1
+		idx.tests[i] = test.Condition
+		paths[i] = test.SubPath
+	}
+	// Check for conflicts.
 	for _, test := range tests {
 		i := len(test.SubPath.Nodes) - 1
+		// Check that a single node must not be marked both MustExist and MustNotExist.
 		v, ok := idx.tests[i]
 		if ok && v != test.Condition {
-			return nil, errors.New("conflicting path test conditions")
+			return nil, fmt.Errorf("%w: path %q is marked both MustExist and MustNotExist", ErrConflict, test.SubPath)
+		}
+
+		if v != MustExist {
+			continue
+		}
+		// Since this node MustExist, check all parents to ensure none of them MustNotExist.
```

Comment:
What if we stored the index of the lowest `mustNotExist` test we encounter and the highest `mustExist` test we encounter and make sure `mustExist highest` < `mustNotExist  lowest`? That would avoid an n^2 comparison and storing the map in memory.

Reply:
Good idea. Done!